plugins/lookup/hashi_vault.py

```diff
@@ -428,6 +440,22 @@ def auth_aws_iam_login(self):
             Display().warning("HVAC should be updated to version 0.9.3 or higher. Deprecated method 'auth_aws_iam' will be used.")
             self.client.auth_aws_iam(**params)
 
+    def auth_jwt(self, **kwargs):
+        role_id = kwargs.get('role_id', os.environ.get('VAULT_ROLE_ID', None))
+        if role_id is None:
+            raise AnsibleError("Authentication method jwt requires a role_id")
+
+        jwt = kwargs.get('jwt', os.environ.get('VAULT_JWT', None))
+        if jwt is None:
+            raise AnsibleError("Authentication method jwt requires parameter jwt to be set")
+
+        mount_point = kwargs.get('mount_point')
+        if mount_point is None:
+            mount_point = 'jwt'
+
+        # The GCP (and Azure, Kubernetes) auth methods just implement JWT
+        self.client.auth.gcp.login(role=role_id, jwt=jwt, mount_point=mount_point)
```

Comment:
As discussed in #154 , I don't think we should overload a specific JWT implementation like GCP and call it JWT in the Ansible plugin. If you'd like to add GCP (named as such), then we can note in the documentation that as of that writing it may be used for JWT auth, but we should be clear that it's a GCP auth method and that its continued compatibility with other JWT applications is not guranteed.

However it does seem that HVAC has added generic JWT/OIDC paths; if so we should use those (this would also require error checking for the HVAC version, like we do for other auth methods). See my comment on https://github.com/ansible-collections/community.general/pull/154#issuecomment-719729868

Reply:
I am looking at having `auth_gcp` as the fallback method, but using the new (since hvac 0.10.5) method `auth.jwt.jwt_login`. Does that sound ok to you?

Reply:
@briantist I am now using the `auth_gcp` only as a fallback method. Please have another look...